.pre-commit-config.yaml

```diff
@@ -0,0 +1,26 @@
+repos:
+- repo: https://github.com/pre-commit/pre-commit-hooks
+  rev: v4.4.0
+  hooks:
+  - id: trailing-whitespace
+  - id: end-of-file-fixer
+  - id: check-yaml
+  - id: debug-statements
+- repo: https://github.com/pycqa/flake8.git
+  rev: 6.0.0
+  hooks:
+  - id: flake8
+    args: [--max-line-length, "100", --ignore, "E128,E811,W503,E203"]
+- repo: https://github.com/asottile/pyupgrade
+  rev: v3.7.0
+  hooks:
+  - id: pyupgrade
+    args: [--py38-plus, --keep-runtime-typing]
+- repo: https://github.com/shellcheck-py/shellcheck-py
+  rev: v0.9.0.5
+  hooks:
+  - id: shellcheck
+- repo: https://github.com/AleksaC/hadolint-py
+  rev: v2.12.0.2
+  hooks:
+  - id: hadolint
```

Comment:
Thanks for adding this - this is for Dockerfiles, which we don't have yet, but I appreciate you adding it in advance :+1: